tests/cephadm/test_os_tuning_profile.py

```diff
@@ -0,0 +1,113 @@
+import tempfile
+from copy import deepcopy
+
+import yaml
+
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class OsTuningProfileError(Exception):
+    pass
+
+
+def generate_tuned_profile_spec(installler_node, specs):
+    """
+    Generate tuned profile.
+    Args:
+        installler_node: installer node
+        specs: tuned profile specs
+    """
+
+    log.info(f"Spec yaml file content:\n{specs}")
+    # Create tuned profile spec yaml file
+    temp_file = tempfile.NamedTemporaryFile(suffix=".yaml")
+    spec_file = installler_node.remote_file(
+        sudo=True, file_name=temp_file.name, file_mode="wb"
+    )
+    spec = yaml.dump(specs, sort_keys=False, indent=2).encode("utf-8")
+    spec_file.write(spec)
+    spec_file.flush()
+    return temp_file.name
+
+
+def run(ceph_cluster, **kw):
+    """Verify tuning profile test cases.
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+        kw: test data
+        e.g:
+        test:
+            name: create_tuning_profile
+            desc: Verify tuning profile created successfully using cephadm
+            polarion-id: CEPH-83575318
+            module: test_os_tuning_profile.py
+            config:
+                steps:
+                - config:
+                    command: tuned_profile_apply
+                    specs:
+                    profile_name: test-mon-host-profile
+                    placement:
+                        hosts:
+                        - node1
+                        - node2
+                    settings:
+                        fs.file-max: 100000
+                        vm.swappiness: 14
+    """
+    installler_node = ceph_cluster.get_nodes(role="installer")[0]
+    config = deepcopy(kw["config"])
```

Comment:
Why do we need to have deepcopy of kw ?? Can't be `kw` used directly ?

Reply:
Updated